### PR TITLE
CLI: Make vite the default builder

### PR DESCRIPTION
**Pull request by kasperpeulen**
## What I did

New projects without a build setup still default to webpack. However, as vite is less prone to config problems (like a missing babelrc file) and in many cases faster, I think that it would makes sense to make vite the default now. 

## How to test

Create a new project, and run npx sb init, and see that it will get a vite builder.

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
This needs thorough QA!

**Comment by joeycozza** [comment]
Seems like you'd also want to update documentation around this at the same time.

https://github.com/storybookjs/storybook/blob/next/docs/builders/webpack.md?plain=1#L5

> Storybook Webpack builder is the default builder for Storybook.

**Comment by jonniebigodes** [comment]
@joeycozza it will be. Once this is merged I'll push a pull request adjusting it, thanks for the heads up. Appreciate it 🙏

**Comment by shilman** [comment]
Let's wait to patch this back until it's been thoroughly tested.

File: code/lib/cli/src/detect.ts [hunk 1]
```diff
@@ -123,14 +123,12 @@ export function detectBuilder(packageManager: JsPackageManager, projectType: Pro
...
-    case ProjectType.SVELTE:
-    case ProjectType.SVELTEKIT:
-    case ProjectType.VUE:
-    case ProjectType.VUE3:
-    case ProjectType.SFC_VUE:
-      return CoreBuilder.Vite;
-    default:
+    case ProjectType.NEXTJS:
+    case ProjectType.REACT_SCRIPTS:
+    case ProjectType.WEBPACK_REACT:
```
Comment (yannbf) [inline code/lib/cli/src/detect.ts:128]: I think you'd need to add more stuff here, such as: Angular, Aurelia, Ember, Marionette, Marko, Mithril, Rax, React_project, Riot, Server.

**Review by yannbf — CHANGES_REQUESTED** [review]
Apart from doing thorough QA, it's important to note that differently than builder-webpack, which has webpack as a dependency, the builder-vite has vite as peer dependency. This means projects won't really work unless we made bigger changes to the CLI, to install Vite as well!

**Comment by ndelangen** [comment]
Fair point @yannbf
So the CLI should add vite as a dependency for the user, if the framework chosen is vite-based, and no vite dependency is found.

This might not be a really complex change, @kasperpeulen are you planning to make this change?

**Comment by yannbf** [comment]
Given the behavior was changed here: https://github.com/storybookjs/storybook/pull/22492
I believe we can safely close this PR!

**Pull request closed without merging**